pkg/catalogsourceconfig/registry.go

```diff
@@ -356,6 +375,34 @@ func (r *registry) newServiceSpec() core.ServiceSpec {
 	}
 }
 
+// waitForDeploymentScaleDown waits for the deployment to scale down to zero within the timeout duration.
+func (r *registry) waitForDeploymentScaleDown(retryInterval, timeout time.Duration) (*apps.Deployment, error) {
+	deployment := apps.Deployment{}
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		err = r.client.Get(context.TODO(), r.csc.key(), &deployment)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				r.log.Infof("Waiting for availability of %s Deployment", deployment.GetName())
+				return false, nil
```

Comment:
Question: If the deployment object is not found is it worth waiting?

Reply:
This is from the SDK code. I assume this won't happen in our case. But I see no harm in leaving this in.

Reply:
But I am open to removing it too.

Reply:
Fixed.